src/App.js

```diff
@@ -1,24 +1,73 @@
-import React from "react";
-import logo from "./logo.svg";
-import "./App.css";
-import Button from "./Button.js";
-import Display from "./Display.js";
-import constant from "./constants";
-import "./Style.css";
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+import Button from './button';
+import Display from './display';
+import { BUTTON_PROPERTIES_ARRAY } from './constants';
+import { updateResult } from './pressButtonLogic';
 
-const App = () => {
-  let index = 0;
-  let arr = constant.arr;
-  return (
-    <>
-      <Display />
-      <div className="ButtonsArea">
-        {arr.map(value => (
-          <Button key={value} index={index++} value={value} />
-        ))}
-      </div>
-    </>
-  );
-};
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            result: ['0'],
+            operators: {
+                array: [],
+                redundant: 0,
+            },
+            default: true,
+            point: false,
+        };
+        this.updateState = this.updateState.bind(this);
+    }
 
+    updateState(value) {
+        this.setState({ ...updateResult(value, this.state) });
+    }
+
+    getButtonProperties(properties, index) {
+        let newProperties = { ...BUTTON_PROPERTIES_ARRAY[index] };
+        if (index === 0) {
+            if (
+                this.state.default === false ||
+                this.state.operators.array.length > 0
+            ) {
+                newProperties.value = 'C';
+            } else {
+                newProperties.value = 'AC';
+            }
+        }
+
+        return newProperties;
+    }
+
+    render() {
+        return (
+            <>
+                <Display
+                    value={this.state.result[this.state.result.length - 1]}
+                />
+                <div
+                    className="buttons-area"
+                    style={{ width: '240px', height: '250px' }}
+                >
+                    {BUTTON_PROPERTIES_ARRAY.map((properties, index) => {
+                        let buttonProperties = this.getButtonProperties(
+                            properties,
+                            index
+                        );
+                        return (
+                            <Button
+                                key={buttonProperties.value}
+                                value={buttonProperties.value}
+                                properties={buttonProperties}
```

Comment:
The Button component needs only `value` and `updateState`. Why do you pass `properties` too?